homeassistant/components/venstar/sensor.py

```diff
@@ -78,58 +120,61 @@ def unique_id(self):
         """Return the unique id."""
         return f"{self._config.entry_id}_{self.sensor_name.replace(' ', '_')}_{self.entity_description.key}"
 
-
-class VenstarHumiditySensor(VenstarSensor):
-    """Represent a Venstar humidity sensor."""
-
     @property
     def name(self):
         """Return the name of the device."""
-        return f"{self._client.name} {self.sensor_name} Humidity"
+        return self.entity_description.name_fn(self.coordinator, self.sensor_name)
 
     @property
     def native_value(self) -> int:
         """Return state of the sensor."""
-        return self._client.get_sensor(self.sensor_name, "hum")
-
-
-class VenstarTemperatureSensor(VenstarSensor):
-    """Represent a Venstar temperature sensor."""
-
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return (
-            f"{self._client.name} {self.sensor_name.replace(' Temp', '')} Temperature"
-        )
+        return self.entity_description.value_fn(self.coordinator, self.sensor_name)
 
     @property
     def native_unit_of_measurement(self) -> str:
         """Return unit of measurement the value is expressed in."""
-        if self._client.tempunits == self._client.TEMPUNITS_F:
-            return TEMP_FAHRENHEIT
-        return TEMP_CELSIUS
-
-    @property
-    def native_value(self) -> float:
-        """Return state of the sensor."""
-        return round(float(self._client.get_sensor(self.sensor_name, "temp")), 1)
+        return self.entity_description.uom_fn(self.coordinator)
 
 
 SENSOR_ENTITIES: tuple[VenstarSensorEntityDescription, ...] = (
     VenstarSensorEntityDescription(
-        key="humidity",
+        key="hum",
         device_class=DEVICE_CLASS_HUMIDITY,
         state_class=STATE_CLASS_MEASUREMENT,
-        native_unit_of_measurement=PERCENTAGE,
-        cls=VenstarHumiditySensor,
-        stype="hum",
+        uom_fn=lambda coordinator: PERCENTAGE,
```

Comment:
Since the unit is always a constant we can just use the attribute `native_unit_of_measurement` and set it to the constant. We can remove `uom_fn` from the description class.

Reply:
I had to do this, because the temperature one has a complex uom.  I'm not sure how to deal with it without a separate class if I don't use the uom_fn thing here.

Reply:
Right, I missed that one.